smtp/client.js

```diff
@@ -85,7 +85,7 @@ Client.prototype =
       if(!(msg instanceof message.Message) 
           && msg.from 
           && (msg.to || msg.cc || msg.bcc)
-          && (msg.text || this._containsInlinedHtml(msg.attachment)))
+          && ((msg.text || '' != null) || this._containsInlinedHtml(msg.attachment)))
```

Comment:
I'm not sure. There is already an undefined check using the double pipe.

```
//msg.text or blank is not null
msg.text || '' != null
```

In this case `msg.text` is checked for `undefined` and `null` at the same time.

It would probably be worth double checking either way, I think in the end I got so confused it was the only method which handled all scenarios and didn't break other unit tests!